src/io/avro/write/schema.rs

```diff
@@ -8,10 +8,11 @@ use crate::error::{Error, Result};
 
 /// Converts a [`Schema`] to an Avro [`Record`].
 pub fn to_record(schema: &Schema) -> Result<Record> {
+    let mut name_counter: i32 = 0;
     let fields = schema
         .fields
         .iter()
-        .map(field_to_field)
+        .map(|f| field_to_field(&f, &mut name_counter))
```

Comment:
won't this be unstable on addition of a new field? Could the counter be reset between fields, so that a field's name is independent of previous fields in the schema?

Reply:
Yes, perhaps you could, it wouldn't be a reset since needs to be globally unique but you could maybe increment the counter by some large number (would be weird). But imo it shouldn't be a goal to rely on the record names when converting an Arrow schema to an Avro schema.